pdf-text-spring-boot-starter fails to initialize

The spring boot starter for pdf-text fails to initialize.

```
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'pdfController' defined in file [C:\Users\Cody\repos\xpdf-web\target\classes\io\xpdf\web\api\pdf\PdfController.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'pdfTextTool' defined in class path resource [io/xpdf/api/pdftext/autoconfigure/PdfTextToolAutoConfiguration.class]: Failed to instantiate [io.xpdf.api.pdftext.PdfTextTool]: Factory method 'pdfTextTool' threw exception with message: The configured executable does not exist
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:798) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:237) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1354) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1191) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:561) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:521) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:325) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:323) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:975) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:959) ~[spring-context-6.1.3.jar:6.1.3]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:624) ~[spring-context-6.1.3.jar:6.1.3]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.2.2.jar:3.2.2]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-3.2.2.jar:3.2.2]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456) ~[spring-boot-3.2.2.jar:3.2.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:334) ~[spring-boot-3.2.2.jar:3.2.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1354) ~[spring-boot-3.2.2.jar:3.2.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1343) ~[spring-boot-3.2.2.jar:3.2.2]
	at io.xpdf.web.SpringBootRestApplication.main(SpringBootRestApplication.java:10) ~[classes/:na]
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'pdfTextTool' defined in class path resource [io/xpdf/api/pdftext/autoconfigure/PdfTextToolAutoConfiguration.class]: Failed to instantiate [io.xpdf.api.pdftext.PdfTextTool]: Factory method 'pdfTextTool' threw exception with message: The configured executable does not exist
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:651) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:639) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1334) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1164) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:561) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:521) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:325) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:323) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1443) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1353) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:907) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:785) ~[spring-beans-6.1.3.jar:6.1.3]
	... 19 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.xpdf.api.pdftext.PdfTextTool]: Factory method 'pdfTextTool' threw exception with message: The configured executable does not exist
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:177) ~[spring-beans-6.1.3.jar:6.1.3]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:647) ~[spring-beans-6.1.3.jar:6.1.3]
	... 33 common frames omitted
Caused by: io.xpdf.api.common.exception.XpdfRuntimeException: The configured executable does not exist
	at io.xpdf.api.pdftext.PdfTextTool$PdfTextToolBuilder.configureExecutableFile(PdfTextTool.java:116) ~[pdf-text-api-1.0.0.jar:na]
	at io.xpdf.api.pdftext.PdfTextTool$PdfTextToolBuilder.build(PdfTextTool.java:89) ~[pdf-text-api-1.0.0.jar:na]
	at io.xpdf.api.pdftext.autoconfigure.PdfTextToolAutoConfiguration.pdfTextTool(PdfTextToolAutoConfiguration.java:53) ~[pdf-text-api-spring-boot-starter-1.0.0.jar:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:140) ~[spring-beans-6.1.3.jar:6.1.3]
	... 34 common frames omitted
```

--- a/common-api/src/test/kotlin/io/xpdf/api/common/util/XpdfUtilsTest.kt
+++ b/common-api/src/test/kotlin/io/xpdf/api/common/util/XpdfUtilsTest.kt
@@ -19,7 +19,7 @@ package io.xpdf.api.common.util
 import io.kotest.assertions.throwables.shouldThrowWithMessage
 import io.kotest.matchers.shouldBe
 import io.mockk.mockkStatic
-import io.mockk.unmockkStatic
+import io.mockk.unmockkAll
 import io.xpdf.api.common.exception.XpdfRuntimeException
 import org.junit.jupiter.api.AfterEach
 import org.junit.jupiter.api.BeforeEach
@@ -41,7 +41,7 @@ class XpdfUtilsTest {
 
     @AfterEach
     fun afterEach() {
-        unmockkStatic(XpdfUtils::class)
+        unmockkAll()
 
         // reset system properties that test may have altered
         System.setProperty("java.io.tmpdir", javaTmpDir)

--- a/pdf-text-api-spring-boot-starter/src/main/java/io/xpdf/api/pdftext/autoconfigure/PdfTextToolAutoConfiguration.java
+++ b/pdf-text-api-spring-boot-starter/src/main/java/io/xpdf/api/pdftext/autoconfigure/PdfTextToolAutoConfiguration.java
@@ -17,14 +17,11 @@
 package io.xpdf.api.pdftext.autoconfigure;
 
 import io.xpdf.api.pdftext.PdfTextTool;
-import io.xpdf.api.pdftext.util.PdfTextUtils;
 import org.springframework.boot.autoconfigure.AutoConfiguration;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
-
-import java.nio.file.Path;
 
 /**
  * An autoconfiguration for {@link PdfTextTool} with configuration properties defined in {@link PdfTextToolProperties}.
@@ -39,17 +36,9 @@ public class PdfTextToolAutoConfiguration {
     @Bean
     @ConditionalOnMissingBean
     public PdfTextTool pdfTextTool(PdfTextToolProperties pdfTextToolProperties) {
-        Path executablePath = pdfTextToolProperties.getExecutablePath() == null
-                ? PdfTextUtils.getPdfTextExecutablePath()
-                : pdfTextToolProperties.getExecutablePath();
-
-        Integer timeoutSeconds = pdfTextToolProperties.getTimeoutSeconds() == null
-                ? PdfTextUtils.getPdfTextTimeoutSeconds()
-                : pdfTextToolProperties.getTimeoutSeconds();
-
         return PdfTextTool.builder()
-                .executableFile(executablePath.toFile())
-                .timeoutSeconds(timeoutSeconds)
+                .executableFile(pdfTextToolProperties.getExecutablePath() != null ? pdfTextToolProperties.getExecutablePath().toFile() : null)
+                .timeoutSeconds(pdfTextToolProperties.getTimeoutSeconds())
                 .build();
     }
 

--- a/pdf-text-api-spring-boot-starter/src/test/kotlin/io/xpdf/api/pdftext/autoconfigure/PdfTextToolAutoConfigurationTest.kt
+++ b/pdf-text-api-spring-boot-starter/src/test/kotlin/io/xpdf/api/pdftext/autoconfigure/PdfTextToolAutoConfigurationTest.kt
@@ -17,12 +17,12 @@
 package io.xpdf.api.pdftext.autoconfigure
 
 import io.kotest.matchers.shouldBe
-import io.mockk.every
 import io.mockk.mockk
-import io.mockk.mockkStatic
-import io.mockk.unmockkStatic
+import io.xpdf.api.common.util.XpdfUtils
 import io.xpdf.api.pdftext.PdfTextTool
 import io.xpdf.api.pdftext.util.PdfTextUtils
+import org.apache.commons.io.FileUtils
+import org.junit.jupiter.api.AfterAll
 import org.junit.jupiter.api.AfterEach
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
@@ -31,12 +31,21 @@ import org.springframework.context.annotation.AnnotationConfigApplicationContext
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import java.io.File
-import java.nio.file.Path
-import java.nio.file.Paths
+import kotlin.io.path.pathString
 
 class PdfTextToolAutoConfigurationTest {
 
     private var context: AnnotationConfigApplicationContext = AnnotationConfigApplicationContext()
+
+    companion object {
+        private val pdfTextToolBean = mockk<PdfTextTool>()
+
+        @JvmStatic
+        @AfterAll
+        fun afterAll() {
+            FileUtils.deleteQuietly(XpdfUtils.getXpdfTempPath().toFile())
+        }
+    }
 
     @BeforeEach
     fun beforeEach() {
@@ -49,9 +58,9 @@ class PdfTextToolAutoConfigurationTest {
     }
 
     @Test
-    fun `should autoconfigure pdf text tool from properties`() {
+    fun `should autoconfigure pdf text tool with properties`() {
         // given
-        val executableFile: File = Paths.get(System.getProperty("java.io.tmpdir"), "executableName").toFile().apply {
+        val executableFile: File = XpdfUtils.getXpdfTempPath().resolve("executableName").toFile().apply {
             createNewFile()
             deleteOnExit()
         }
@@ -73,20 +82,8 @@ class PdfTextToolAutoConfigurationTest {
     }
 
     @Test
-    fun `should autoconfigure pdf text tool from xpdf utils`() {
+    fun `should autoconfigure pdf text tool without properties`() {
         // given
-        val executableFile = mockk<File> {
-            every { exists() } returns true
-            every { setExecutable(any()) } returns true
-        }
-        val executablePath = mockk<Path> {
-            every { toFile() } returns executableFile
-        }
-
-        mockkStatic(PdfTextUtils::class)
-        every { PdfTextUtils.getPdfTextExecutablePath() } returns executablePath
-        every { PdfTextUtils.getPdfTextTimeoutSeconds() } returns 99
-
         context.register(PdfTextToolAutoConfiguration::class.java)
         context.refresh()
 
@@ -94,10 +91,8 @@ class PdfTextToolAutoConfigurationTest {
         val pdfTextTool = context.getBean(PdfTextTool::class.java)
 
         // then
-        pdfTextTool.executableFile shouldBe executableFile
-        pdfTextTool.timeoutSeconds shouldBe 99
-
-        unmockkStatic(PdfTextUtils::class)
+        pdfTextTool.executableFile.canonicalPath shouldBe PdfTextUtils.getPdfTextExecutablePath().pathString
+        pdfTextTool.timeoutSeconds shouldBe PdfTextUtils.getPdfTextTimeoutSeconds()
     }
 
     @Test
@@ -108,10 +103,6 @@ class PdfTextToolAutoConfigurationTest {
 
         // when then
         context.getBean(PdfTextTool::class.java) shouldBe pdfTextToolBean
-    }
-
-    companion object {
-        private val pdfTextToolBean = mockk<PdfTextTool>()
     }
 
     @Configuration

--- a/pdf-text-api-spring-boot-starter/src/test/kotlin/io/xpdf/api/pdftext/autoconfigure/PdfTextToolAutoConfigurationTest.kt
+++ b/pdf-text-api-spring-boot-starter/src/test/kotlin/io/xpdf/api/pdftext/autoconfigure/PdfTextToolAutoConfigurationTest.kt
@@ -62,6 +62,7 @@ class PdfTextToolAutoConfigurationTest {
     fun `should autoconfigure pdf text tool with properties`() {
         // given
         val executableFile: File = XpdfUtils.getXpdfTempPath().resolve("executableName").toFile().apply {
+            mkdirs()
             createNewFile()
             deleteOnExit()
         }

--- a/pdf-text-api-spring-boot-starter/src/test/kotlin/io/xpdf/api/pdftext/autoconfigure/PdfTextToolAutoConfigurationTest.kt
+++ b/pdf-text-api-spring-boot-starter/src/test/kotlin/io/xpdf/api/pdftext/autoconfigure/PdfTextToolAutoConfigurationTest.kt
@@ -17,7 +17,7 @@
 package io.xpdf.api.pdftext.autoconfigure
 
 import io.kotest.matchers.shouldBe
-import io.mockk.mockk
+import io.mockk.*
 import io.xpdf.api.common.util.XpdfUtils
 import io.xpdf.api.pdftext.PdfTextTool
 import io.xpdf.api.pdftext.util.PdfTextUtils
@@ -31,7 +31,7 @@ import org.springframework.context.annotation.AnnotationConfigApplicationContext
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import java.io.File
-import kotlin.io.path.pathString
+import java.nio.file.Path
 
 class PdfTextToolAutoConfigurationTest {
 
@@ -54,6 +54,7 @@ class PdfTextToolAutoConfigurationTest {
 
     @AfterEach
     fun afterEach() {
+        unmockkAll()
         context.close()
     }
 
@@ -84,6 +85,18 @@ class PdfTextToolAutoConfigurationTest {
     @Test
     fun `should autoconfigure pdf text tool without properties`() {
         // given
+        val executableFile = mockk<File> {
+            every { exists() } returns true
+            every { setExecutable(any()) } returns true
+        }
+        val executablePath = mockk<Path> {
+            every { toFile() } returns executableFile
+        }
+
+        mockkStatic(PdfTextUtils::class)
+        every { PdfTextUtils.getPdfTextExecutablePath() } returns executablePath
+        every { PdfTextUtils.getPdfTextTimeoutSeconds() } returns 99
+
         context.register(PdfTextToolAutoConfiguration::class.java)
         context.refresh()
 
@@ -91,8 +104,10 @@ class PdfTextToolAutoConfigurationTest {
         val pdfTextTool = context.getBean(PdfTextTool::class.java)
 
         // then
-        pdfTextTool.executableFile.canonicalPath shouldBe PdfTextUtils.getPdfTextExecutablePath().pathString
-        pdfTextTool.timeoutSeconds shouldBe PdfTextUtils.getPdfTextTimeoutSeconds()
+        pdfTextTool.executableFile shouldBe executableFile
+        pdfTextTool.timeoutSeconds shouldBe 99
+
+        unmockkStatic(PdfTextUtils::class)
     }
 
     @Test

--- a/pdf-text-api/src/main/java/io/xpdf/api/pdftext/PdfTextTool.java
+++ b/pdf-text-api/src/main/java/io/xpdf/api/pdftext/PdfTextTool.java
@@ -286,7 +286,7 @@ public class PdfTextTool implements XpdfTool<PdfTextRequest, PdfTextResponse> {
         }
 
         // create output directory if not exists
-        textFile.getParentFile().mkdir();
+        textFile.getParentFile().mkdirs();
 
         return textFile;
     }

--- a/pdf-text-api/src/test/kotlin/io/xpdf/api/pdftext/PdfTextToolCucumberIT.kt
+++ b/pdf-text-api/src/test/kotlin/io/xpdf/api/pdftext/PdfTextToolCucumberIT.kt
@@ -16,7 +16,10 @@
  */
 package io.xpdf.api.pdftext
 
+import io.cucumber.java.After
 import io.cucumber.junit.platform.engine.Constants.GLUE_PROPERTY_NAME
+import io.xpdf.api.common.util.XpdfUtils
+import org.apache.commons.io.FileUtils
 import org.junit.platform.suite.api.ConfigurationParameter
 import org.junit.platform.suite.api.SelectClasspathResource
 import org.junit.platform.suite.api.Suite
@@ -24,4 +27,11 @@ import org.junit.platform.suite.api.Suite
 @Suite
 @SelectClasspathResource("io/xpdf/api/pdftext")
 @ConfigurationParameter(key = GLUE_PROPERTY_NAME, value = "io.xpdf.api.pdftext"  )
-class PdfTextToolCucumberIT
+class PdfTextToolCucumberIT {
+
+    @After
+    fun after() {
+        FileUtils.deleteQuietly(XpdfUtils.getXpdfTempPath().toFile())
+    }
+
+}

--- a/pdf-text-api/src/test/kotlin/io/xpdf/api/pdftext/PdfTextToolCucumberSteps.kt
+++ b/pdf-text-api/src/test/kotlin/io/xpdf/api/pdftext/PdfTextToolCucumberSteps.kt
@@ -25,21 +25,32 @@ import io.kotest.matchers.shouldBe
 import io.kotest.matchers.shouldNotBe
 import io.kotest.matchers.string.shouldMatch
 import io.xpdf.api.common.exception.XpdfException
+import io.xpdf.api.common.util.XpdfUtils
 import io.xpdf.api.pdftext.options.PdfTextEncoding
 import io.xpdf.api.pdftext.options.PdfTextEndOfLine
 import io.xpdf.api.pdftext.options.PdfTextFormat
 import io.xpdf.api.pdftext.util.PdfTextUtils
 import org.apache.commons.io.FileUtils
+import org.springframework.test.context.event.annotation.AfterTestClass
 import java.io.File
 import java.nio.file.Paths
 
 class PdfTextToolCucumberSteps {
+
     private var toolDto: PdfTextToolDto? = null
     private var requestDto: PdfTextRequestDto? = null
     private var optionsDto: PdfTextOptionsDto? = null
     private var nativeOptionsDto: NativeOptionsDto? = null
     private var response: PdfTextResponse? = null
     private var exception: XpdfException? = null
+
+    companion object {
+        @JvmStatic
+        @AfterTestClass
+        fun afterAll() {
+            FileUtils.deleteQuietly(XpdfUtils.getXpdfTempPath().toFile())
+        }
+    }
 
     @Given("a PdfTextTool")
     fun `a PdfTextTool`() {
@@ -59,7 +70,7 @@ class PdfTextToolCucumberSteps {
     @Given("a PdfTextTool with {int} second timeout and dynamic executable file")
     fun `a PdfTextTool with TIMEOUT_SECONDS second timeout and dynamic executable file`(timeoutSeconds: Int) {
         val executableResourceStream = this::class.java.classLoader.getResourceAsStream(PdfTextUtils.getPdfTextExecutableResourceName())!!
-        val executableFile = Paths.get(System.getProperty("java.io.tmpdir")).resolve("some.exe").toFile()
+        val executableFile = XpdfUtils.getXpdfTempPath().resolve("some.exe").toFile()
         FileUtils.copyInputStreamToFile(executableResourceStream, executableFile)
         executableFile.setExecutable(true)
 
@@ -79,7 +90,7 @@ class PdfTextToolCucumberSteps {
     @Given("a PdfTextRequest with pdf file {word} and dynamic text file")
     fun `a PdfTextRequest with pdf file PDF_FILE_NAME and dynamic text file`(pdfFileName: String) {
         val pdfFile = File(this.javaClass.classLoader.getResource("pdfs/${pdfFileName}")!!.toURI())
-        val txtFile = Paths.get(System.getProperty("java.io.tmpdir")).resolve("some.txt").toFile()
+        val txtFile = XpdfUtils.getXpdfTempPath().resolve("some.txt").toFile()
 
         this.requestDto = PdfTextRequestDto(pdfFile, txtFile)
     }
@@ -146,7 +157,7 @@ class PdfTextToolCucumberSteps {
     @DataTableType
     fun pdfTextRequestDtoTransformer(row: Map<String, String?>) = PdfTextRequestDto(
         File(this.javaClass.classLoader.getResource("pdfs/${row["pdfFile"]!!}")!!.toURI()),
-        row["textFile"]?.let { Paths.get(System.getProperty("java.io.tmpdir")).resolve(it).toFile() },
+        row["textFile"]?.let { XpdfUtils.getXpdfTempPath().resolve(it).toFile() },
     )
 
     @DataTableType

--- a/pdf-text-api/src/test/kotlin/io/xpdf/api/pdftext/PdfTextToolTest.kt
+++ b/pdf-text-api/src/test/kotlin/io/xpdf/api/pdftext/PdfTextToolTest.kt
@@ -26,12 +26,15 @@ import io.kotest.matchers.string.shouldContain
 import io.kotest.matchers.string.shouldMatch
 import io.mockk.*
 import io.xpdf.api.common.exception.*
+import io.xpdf.api.common.util.XpdfUtils
 import io.xpdf.api.pdftext.options.PdfTextEncoding
 import io.xpdf.api.pdftext.options.PdfTextEndOfLine
 import io.xpdf.api.pdftext.options.PdfTextFormat
 import io.xpdf.api.pdftext.util.PdfTextUtils
 import org.apache.commons.io.FileUtils
 import org.apache.commons.io.IOUtils
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.AfterEach
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.extension.ExtendWith
 import org.junit.jupiter.params.ParameterizedTest
@@ -51,6 +54,19 @@ class PdfTextToolTest {
 
     private val pdfTextTool = PdfTextTool.builder().build()
 
+    companion object {
+        @JvmStatic
+        @AfterAll
+        fun afterAll() {
+            FileUtils.deleteQuietly(XpdfUtils.getXpdfTempPath().toFile())
+        }
+    }
+
+    @AfterEach
+    fun afterEach() {
+        unmockkAll()
+    }
+
     @Test
     fun `should initialize and copy executable to local system`() {
         // given
@@ -68,9 +84,6 @@ class PdfTextToolTest {
 
         // then
         verify { FileUtils.copyInputStreamToFile(any(), any()) }
-
-        unmockkStatic(PdfTextUtils::class)
-        unmockkStatic(FileUtils::class)
     }
 
     @Test
@@ -89,9 +102,6 @@ class PdfTextToolTest {
 
         // then
         verify(exactly = 0) { FileUtils.copyInputStreamToFile(any(), any()) }
-
-        unmockkStatic(PdfTextUtils::class)
-        unmockkStatic(FileUtils::class)
     }
 
     @Test
@@ -123,8 +133,6 @@ class PdfTextToolTest {
         shouldThrowWithMessage<XpdfRuntimeException>("Unable to locate executable in project resources") {
             PdfTextTool.builder().build()
         }
-
-        unmockkStatic(PdfTextUtils::class)
     }
 
     @Test
@@ -143,9 +151,6 @@ class PdfTextToolTest {
         shouldThrowWithMessage<XpdfRuntimeException>("Unable to copy executable from resources to local system") {
             PdfTextTool.builder().build()
         }
-
-        unmockkStatic(PdfTextUtils::class)
-        unmockkStatic(FileUtils::class)
     }
 
     @Test
@@ -161,8 +166,6 @@ class PdfTextToolTest {
         shouldThrowWithMessage<XpdfRuntimeException>("Unable to set execute permissions on executable") {
             PdfTextTool.builder().build()
         }
-
-        unmockkStatic(PdfTextUtils::class)
     }
 
     @Test
@@ -203,8 +206,6 @@ class PdfTextToolTest {
 
         // then
         result.timeoutSeconds shouldBe 99
-
-        unmockkStatic(PdfTextUtils::class)
     }
 
     @Test
@@ -257,9 +258,6 @@ class PdfTextToolTest {
         capturedOutput.all shouldContain "Invocation completed; exit code: 0, standard output: standardOutput"
         capturedOutput.all shouldContain "Invocation succeeded"
         capturedOutput.all shouldContain "Process finished"
-
-        unmockkConstructor(ProcessBuilder::class)
-        unmockkStatic(IOUtils::class)
     }
 
     @ParameterizedTest
@@ -313,9 +311,6 @@ class PdfTextToolTest {
         capturedOutput.all shouldContain "Invocation failed; error output: errorOutput"
         capturedOutput.all shouldContain "Process failed; exception message: $message"
         capturedOutput.all shouldContain "Process finished"
-
-        unmockkConstructor(ProcessBuilder::class)
-        unmockkStatic(IOUtils::class)
     }
 
     @Test
@@ -356,9 +351,6 @@ class PdfTextToolTest {
         capturedOutput.all shouldContain "Invocation timed out"
         capturedOutput.all shouldContain "Process failed; exception message: Timeout reached before process could finish"
         capturedOutput.all shouldContain "Process finished"
-
-        unmockkConstructor(ProcessBuilder::class)
-        unmockkStatic(IOUtils::class)
     }
 
     @Test
@@ -509,10 +501,6 @@ class PdfTextToolTest {
         pdfTextTool.initializeTextFile(request) shouldBe textFile
 
         verify { Paths.get("outPath", textFileName) }
-
-        unmockkStatic(PdfTextUtils::class)
-        unmockkStatic(UUID::class)
-        unmockkStatic(Paths::class)
     }
 
     @Test

--- a/pdf-text-api/src/test/kotlin/io/xpdf/api/pdftext/util/PdfTextUtilsTest.kt
+++ b/pdf-text-api/src/test/kotlin/io/xpdf/api/pdftext/util/PdfTextUtilsTest.kt
@@ -19,7 +19,7 @@ package io.xpdf.api.pdftext.util
 import io.kotest.matchers.shouldBe
 import io.mockk.every
 import io.mockk.mockkStatic
-import io.mockk.unmockkStatic
+import io.mockk.unmockkAll
 import io.xpdf.api.common.util.XpdfUtils
 import org.junit.jupiter.api.AfterEach
 import org.junit.jupiter.api.BeforeEach
@@ -38,8 +38,7 @@ class PdfTextUtilsTest {
 
     @AfterEach
     fun afterEach() {
-        unmockkStatic(XpdfUtils::class)
-        unmockkStatic(PdfTextUtils::class)
+        unmockkAll()
     }
 
     @Test

